test/metabase/upload_test.clj

```diff
@@ -112,11 +113,11 @@
 (deftest type-detection-and-parse-test
   (doseq [[string-value expected-value expected-type seps]
           ;; Number-related
-          [["0.0"        0              float-type "."]
-           ["0.0"        0              float-type ".,"]
-           ["0,0"        0              float-type ",."]
-           ["0,0"        0              float-type ", "]
-           ["0.0"        0              float-type ".’"]
+          [["0.0"        0              float-or-int-type "."]
+           ["0.0"        0              float-or-int-type ".,"]
+           ["0,0"        0              float-or-int-type ",."]
+           ["0,0"        0              float-or-int-type ", "]
+           ["0.0"        0              float-or-int-type ".’"]
```

Comment:
Adding this test coverage feels rather orthogonal since we didn't test the old regex either. The change to the regex itself is also trivial. Since you went to the trouble of laying it out though it'll get it over with.

These tests might be good candidates for property based testing, exhausting to maintain otherwise.

Reply:
thanks

Reply:
Interestingly enough it turns out there was actually a preexisting bug here - good instincts. It's not related to the regex though.

| Raw | Parsed | Inserted | Expected  | Correctness
| --- | --- | --- | --- | --- | 
| 0.0 | 0 | 0.0 | 0.0  | 👌 
| (0.0) | 0 | 0.0 | -0.0  | 🙅 
| -0.0 | -0.0 | -0.0 | -0.0 | 👌 

This is an extremely subtle issue, which I'm not sure customers will care about, so unless it's a trivial fix I'll leave for another PR.

Reply:
https://github.com/metabase/metabase/pull/40309